lnwire/dyn_propose.go

```diff
@@ -74,14 +74,24 @@ func (dp *DynPropose) Encode(w *bytes.Buffer, _ uint32) error {
 		return err
 	}
 
-	producers := dynProposeRecords(dp)
+	// Create extra data records.
+	producers, err := dp.ExtraData.RecordProducers()
+	if err != nil {
+		return err
+	}
+
+	// Append the known records.
+	producers = append(producers, dynProposeRecords(dp)...)
 
-	err := EncodeMessageExtraData(&dp.ExtraData, producers...)
+	// Encode all records.
+	var tlvData ExtraOpaqueData
+	err = tlvData.PackRecords(producers...)
 	if err != nil {
 		return err
 	}
 
-	return WriteBytes(w, dp.ExtraData)
+	// Write the records.
```

Comment:
Comments above don't add much, can prob be dropped.

Reply:
Removed